test/e2e_node/cpu_manager_test.go

```diff
@@ -596,17 +597,30 @@ func runCfsQuotaGuPods(ctx context.Context, f *framework.Framework, disabledCPUQ
 	var err error
 	var ctnAttrs []ctnAttribute
 	var pod1, pod2, pod3 *v1.Pod
-	var cleanupPods []*v1.Pod
+	podsToClean := make(map[apitypes.UID]*v1.Pod) // pod.UID -> pod
+
+	cleanupTestPod := func(ctx2 context.Context, pod *v1.Pod) {
```

Comment:
Why restrict its scope to only this function? We could potentially use this in other places as well.

Reply:
right, it's a pattern we have. Added another helper.